src/core/utils/ssrUtils/ssrUtils.ts

```diff
@@ -0,0 +1,8 @@
+class SsrUtils {
+    /**
+     * Checks if code is running on a server environment.
+     */
+    isServer = () => typeof window === 'undefined';
```

Comment:
Very complicated utilities you have here...

Reply:
I love implementing such complex utils :D